src/pages/workspaces/workspace/Tools.js

```diff
@@ -269,11 +288,24 @@ export const Tools = _.flow(
     return _.find({ namespace, name }, configs)
   }
 
+  addRedactedAttribute(config, methods) {
+    const { methodName, methodNamespace, methodVersion, sourceRepo } = config.methodRepoMethod
+    const snapshotObjects = _.map(m => _.pick('snapshotId', m), _.filter({
+      name: methodName,
+      namespace: methodNamespace
+    }, methods))
+    const snapshots = _.map(m => { return m.snapshotId }, snapshotObjects)
+    const isRedacted = !_.includes(methodVersion, snapshots)
+    if (!sourceRepo) reportError('Caller must specify source repo for method')
+    else config.isRedacted = isRedacted
```

Comment:
Please don't mutate data, especially in the render path. It can lead to all sorts of bugs and problems down the line.